rl_power_markets/benchmarks/markets/full.py

```diff
@@ -1,3 +1,220 @@
+            1: {"b":5.5, "CNL":80000, "g_max":6.584, "RU":1.317, "RD":1.317, "CSU":400000, "CSD":80000, "UT":24, "DT":24},
+            2: {"b":30.0, "CNL":6500, "g_max":5.760, "RU":1.152, "RD":1.152, "CSU":32500, "CSD":2850, "UT":20, "DT":20},
+            3: {"b":35.0, "CNL":2851, "g_max":3.781, "RU":1.512, "RD":1.512, "CSU":14250, "CSD":1850, "UT":16, "DT":16},
+            4: {"b":60.0, "CNL":1440, "g_max":3.335, "RU":1.334, "RD":1.334, "CSU":7200, "CSD":1440, "UT":10, "DT":10},
+            5: {"b":80.0, "CNL":1100, "g_max":3.252, "RU":1.951, "RD":1.951, "CSU":5500, "CSD":1200, "UT":8, "DT":8},
+            6: {"b":95.0, "CNL":620, "g_max":2.880, "RU":1.728, "RD":1.728, "CSU":3100, "CSD":1000, "UT":5, "DT":5},
+            7: {"b":100.0, "CNL":560, "g_max":2.748, "RU":2.198, "RD":2.198, "CSU":1120, "CSD":840, "UT":4, "DT":4}
+        }
+        
+        # Setup block structure for each generator
+        self.C = 5  # number of blocks
+        for i in self.generators:
+            step = (self.generators[i]["g_max"] - 0)/self.C
+            blocks = []
+            for b in range(self.C):
+                low = b * step
+                high = (b+1) * step
+                mid = 0.5*(low+high)
+                slope = self.generators[i]["b"] + 2 * 0.0001 * mid
+                cap = high - low
+                blocks.append((low, high, slope, cap))
+            self.generators[i]["blocks"] = blocks
+        
+        # Initialize demand data
+        self.base_demand = sum(gen["g_max"] for gen in self.generators.values()) * 1.0
+        self.demand = []
+        for h in range(self.num_hours):
+            blk = []
+            cap = 10000
+            dec = 5
+            for c in range(self.C):
+                blk.append((cap/self.C, (dec-c)*20))
+            self.demand.append(blk)
+        
+        # Initialize state tensors
+        self.u_i = torch.zeros(self.batch_size, self.num_hours)
+        self.g_i = torch.zeros(self.batch_size, self.num_hours)
+        self.prices = torch.zeros(self.batch_size, self.num_hours)
+        
+        # State space dimensions
+        self.state_dim_per_var = self.num_hours
+        self.num_state_vars = 3  # u_i, g_i, and prices
+        self.obs_size = self.state_dim_per_var * self.num_state_vars
+        self.num_actions = self.num_hours
+        
+        self.num_episodes = 100000
+        self.num_timesteps = 30
+        self.episodes = range(self.num_episodes)
+        self.timesteps = range(self.num_timesteps)
+    
+    def reset(self) -> None:
+        self.u_i.zero_()
+        self.g_i.zero_()
+        self.prices.zero_()
+    
+    def step(self, multipliers: torch.Tensor) -> tuple[torch.Tensor, torch.Tensor]:
+        assert multipliers.shape == (self.batch_size, self.num_hours)
+        assert (multipliers >= 1.0).all()
+        
+        profits = torch.zeros(self.batch_size, 1)
+        
+        # Process each batch item separately since Gurobi doesn't handle batches
```

Comment:
this dependency is the same one they used right? yikes on batch hopefully that won't slow things down too much